include/kokkos-utils/view/slice.hpp

```diff
@@ -0,0 +1,78 @@
+#ifndef KOKKOS_UTILS_VIEW_SLICE_HPP
+#define KOKKOS_UTILS_VIEW_SLICE_HPP
+
+#include "kokkos-utils/concepts/View.hpp"
+
+/**
+ * @file
+ *
+ * This file is a collection of helper functions similar to what can be found in
+ * the standard @c mdspan header, related to retrieving slices of a @c Kokkos::View.
+ *
+ * References:
+ *  - https://en.cppreference.com/w/cpp/container/mdspan
+ *  - https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2023/p2630r4.html
+ */
+
+namespace Kokkos::utils::view
+{
+
+namespace impl
+{
+
+template <typename ViewType, typename TupleA, size_t... A, typename... Indices>
+KOKKOS_FUNCTION
+constexpr auto slice(ViewType&& view, TupleA&& alls, std::index_sequence<A...>, Indices&&... indices) {
```

Comment:
The `typename ViewType` can probably use the concept. We'll also have to think about whether we can find better names for `TupleA` and especially `A`.

Reply:
Perhaps something like an abbreviated `Inds` with a good suffix could be used for `A` and `Indices`?

Reply:
I'm also wondering about the `constexpr`. It looks like in `Kokkos`, the `subview` function is not marked `constexpr`:

- https://github.com/kokkos/kokkos/blob/53ceb4c2d2095fa8173f95652f242f82ca982f12/core/src/View/Kokkos_ViewLegacy.hpp#L1408-L1418

Reply:
Another thought may be whether we should pass the indices by universal reference. Its's just integers, so they may not support move operators. Perhaps just passing by value would be fine for those. May be worth taking a look in `mdspan` to see how it was passed there.

Reply:
If I may one more 😄. The `TupleA` could also become `ArrayA` because we're passing an `std::array`.

Reply:
1. `ViewType` could be constrained with a concept, but considering it is in `impl`, I will not do it. It should reduce compiler pressure probably.
2. `constexpr` is there for the future. When `View` will be a wrapper around `std::mdspan` I expect this to also be used in compile-time expressions.
3. Indices are integers, tuple-like stuff or `Kokkos::ALL`. So I'll keep the forwarding.
4. I'll keep `AllsValues` and `AllsIndices`.

Please have a look at the new version.

Reply:
OK. Just for the use of the `constexpr` for the future, I'm not sure what would be best. Another option would be declare the function without `constexpr` here, and then have a note/fixme in the docstring to say that we expect it to become `constexpr` at some point. An advantage would be that we can keep better track of what's `constexpr` and what's not.

Reply:
I'll keep the `constexpr`.